pgmpy/tests/test_inference/test_ExactInference.py

```diff
@@ -93,7 +93,7 @@ def test_query_multiple_times(self):
                                           np.array([0.772727, 0.227273]))
 
     def test_max_marginal(self):
-        np_test.assert_almost_equal(self.bayesian_inference.max_marginal(), 0.1659, decimal=4)
+        np_test.assert_almost_equal(self.bayesian_inference.max_marginal(['A','R','J','Q','G','L']), 0.0516, decimal=4)
```

Comment:
@lohani2280 Any idea why these values are changing ? And should they ?

Reply:
@ankurankan With  `_variable_elimination` method we are implementing  variable elimination in a generalised way.Many other methods are using this method internally. So, any change here will affect other methods too. 
In our issue we are only concerned with inconsistency with the `query` method.We just want an empty dict to be returned if no variables is specified.I think implementing it locally in the `query` method would be the best idea. I would synchronise the PR soon. Please suggest your views.

Reply:
@lohani2280 I don't think implementing it locally would be a good idea because we have other methods which use `_variable_elimination` and we would like to have the behavior consistent across all these methods. Plus it will add boilerplate code. Could you just check once if the values returned now are correct or the previous values were correct ?

Reply:
@ankurankan The values are actually not changing. If we run the query `bayesian_inference.max_marginal(['A','J','R','Q','L','G'])` , either on this PR or without this, valu is same, i.e, 0.0516. Its just that if we run the query `bayesian_inference.max_marginal()`  on this PR it throws an error because this method call for `_variable_elimination` from which we have just removed the property of returning an empty dict on passing an empty list of variables.

Reply:
@lohani2280 Okay. But isn't calling `max_marginal()` in the previous version and calling `max_marginal(['A', 'J', 'R', 'Q', 'L', 'G'])` in this PR equivalent ? I mean that in the previous version, didn't passing no argument meant computing max_marginal over all the variables in the network ? Because if that's the case the value for your new query should still be `0.1659`. Right ?

Reply:
@ankurankan Right. Let me look a bit more deeply and then i'll get back again.

Reply:
@ankurankan Calling `max_marginal()` in the previous version meant computing max_marginal over all the variables in the network. Also, calling  `max_marginal(['A', 'J', 'R', 'Q', 'L', 'G'])` in the previous version meant computing max_marginal over all the variables in the network. 
In the previous version if we run the query  `bayesian_inference.max_marginal()` , we get `0.1659`. However for `bayesian_inference.max_marginal(['A', 'J', 'R', 'Q', 'L', 'G'])` we get `0.0516`. Ideally, both the query should have same answer.Thus,there is an ambiguity. IMO  value of `max_marginal()` is correct and `max_marginal(['A', 'J', 'R', 'Q', 'L', 'G'])` isn't.
I think i found the bug -- 
When we query `max_marginal(['A', 'J', 'R', 'Q', 'L', 'G'])`, ` _variable_elimination` should return https://github.com/pgmpy/pgmpy/blob/fe2ca2c4ea4d7e1f4a785b9a03a01146826974cf/pgmpy/inference/ExactInference.py#L90
instead of 
https://github.com/pgmpy/pgmpy/blob/fe2ca2c4ea4d7e1f4a785b9a03a01146826974cf/pgmpy/inference/ExactInference.py#L98